src/java.base/unix/native/libnio/ch/Net.c

```diff
@@ -308,6 +312,17 @@ Java_sun_nio_ch_Net_socket0(JNIEnv *env, jclass cl, jboolean preferIPv6,
             close(fd);
             return -1;
         }
+
+        /* Disable IPV6_MULTICAST_ALL if option supported */
+        arg = 0;
+        if ((setsockopt(fd, IPPROTO_IPV6, IPV6_MULTICAST_ALL, (char*)&arg, sizeof(arg)) < 0) &&
+            (errno != ENOPROTOOPT)) {
+            JNU_ThrowByNameWithLastError(env,
+                                     JNU_JAVANETPKG "SocketException",
+                                     "Unable to set IPV6_MULTICAST_ALL");
+            close(fd);
+            return -1;
+        }
```

Comment:
This implementation change looks okay, assuming ENOPROTOOPT on older kernels. I haven't had time to study the test yet, tricky scenario to test reliably.

Reply:
I am running some repeat 50 test runs of the datagram tests on the newer Linux systems.